app/internal/page/payment_confirmation_test.go

```diff
@@ -0,0 +1,280 @@
+package page
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gorilla/sessions"
+
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/pay"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type mockRandom struct {
+	mock.Mock
+}
+
+func (m *mockRandom) String(length int) string {
+	args := m.Called(length)
+	return args.Get(0).(string)
+}
+
+func TestPaymentConfirmation(t *testing.T) {
+	t.Run("Gets payment status from GOV UK Pay by payment_id in cookie and stores payment_id and a UUID against users session ID", func(t *testing.T) {
+		payClient := &mockPayClient{BaseURL: "http://base.url"}
+		ensurePayClientReturnsPaymentInfo(payClient, "abc123")
+
+		w := httptest.NewRecorder()
+
+		template := &mockTemplate{}
+		template.
+			On("Func", w, &paymentConfirmationData{App: appData, PaymentReference: "123456789012"}).
+			Return(nil)
+
+		r, _ := http.NewRequest(http.MethodGet, "/payment-confirmation", nil)
+
+		sessionsStore := &mockSessionsStore{}
+		ensureSessionHasPayCookie(sessionsStore, r)
+		ensureCookieIsExpiredInSession(sessionsStore, r, w)
```

Comment:
Not sure if this makes intent clearer in tests or muddys the water a bit - open to discussion on what others think.

Reply:
That definitely reads better and feels nice to group on the mock object to keep context clear and reduce arguments - I'll update.